Kev/357 import export diary

junhaow1 opened the pull request:
## Pull Request Details

Link to associated issue: https://github.com/anusii/healthpod/issues/357

# Checklist
Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the team style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (make prep or flutter analyze lib)
- [x] Tested on at least one device
  - [x] Windows
- [x] Added 2 reviewers (or 1 for private repositories then they add another)

Commit:
update the diary import and export functionality to match the Appointment data model

Commit:
export working

Commit:
test

Commit:
Merge branch 'dev' into kev/357_import_export_diary

Commit:
Merge branch 'dev' into kev/357_import_export_diary

Commit:
emoved the "isPast" field from both the CSV headers and the processed record.

Commit:
fixed the issue by changing appointmentData['date'] to data['date']

Commit:
fix comments

Commit:
fix import order

Commit:
delete debugPrint

gjwgit reviewed (changes requested):
Testing: created a new appointment but the saved file seems to not include date?

![image](https://github.com/user-attachments/assets/7229f9ec-acee-496c-9628-08e50edf186a)

zheyxu reviewed (changes requested):
Tested on Mac and find data is blank.

<img width="359" alt="Screenshot 2025-05-03 at 12 33 06 am" src="https://github.com/user-attachments/assets/6d4a6e67-3d92-435d-adcf-3b632c6248c8" />

Commit:
Merge branch 'dev' into kev/357_import_export_diary

Commit:
fix date not shown bug

junhaow1 commented:
![image](https://github.com/user-attachments/assets/a42cad24-1f7c-4e5c-a3f4-269e40cce978)

Commit:
handle both cases where the date might be stored and prevent the null error

Commit:
Remove the DIARY tab
Add APPOINTMENT as first feature for VISUAL tab and present a calendar as now in the DIARY tab

Commit:
Add APPOINTMENT as first feature for NEW tab and support creating new appointment

zheyxu reviewed (approved):
Tested and all good.

Commit:
Modified the AppointmentEditorPage to use DataTable directly instead of a separate data table source class, following the pattern used in other editor pages

Commit:
Now users can delete both past and upcoming appointments.

Commit:
fix the "type 'Null' is not a subtype of type 'String'" errors

Commit:
Add APPOINTMENT as first feature for NEW tab and support creating new appointment

Commit:
fix flutter analyze

Commit:
fix import order

Commit:
change code comments

Commit:
add comments

Commit:
add comments

Commit:
modify the _getFormattedDateTime method in date_input.dart to move "at" to the second row

gjwgit committed:
Don't export seconds.

gjwgit committed:
Merge pull request #413 from anusii/kev/387_migrate

Kev/387 migrate to appointment as a feature

Commit:
Merge branch 'dev' into kev/357_import_export_diary
